examples/error_handler.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This program is dedicated to the public domain under the CC0 license.
+
+"""
+This is a very simple example on how one could implement a custom error handler
+"""
+import html
+import logging
+import traceback
+
+from telegram import Update, ParseMode
+from telegram.ext import Updater, CallbackContext, CommandHandler
+
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                    level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+# this can be your own ID, or one for a developer group/channel
+DEVELOPER_CHAT_ID = 208589966
+
+
+def error_handler(update: Update, context: CallbackContext):
+    """Log the error and send a telegram message to notify the developer."""
+    # Log the error before we do anything else, so we can see it even if something breaks.
+    logger.error(msg="Exception while handling an update:", exc_info=context.error)
+
+    # traceback.format_exception returns the usual python message about an exception, but as a
+    # list of strings rather than a single string, so we have to join them together.
+    tb_list = traceback.format_exception(None, context.error, context.error.__traceback__)
+    tb = ''.join(tb_list)
+
+    # Build the message with some markup and additional information about what happened.
+    # You might need to add some logic to deal with messages longer than the 4096 character limit.
+    message = (
+        'An exception was raised while handling an update\n'
+        '<pre>update = {}</pre>\n\n'
+        '<pre>context.chat_data = {}</pre>\n\n'
+        '<pre>context.user_data = {}</pre>\n\n'
+        '<pre>{}</pre>'
+    ).format(
+        html.escape(update.to_json()),
```

Comment:
Can we format the json outputs here for better readability? `json.dumps(update.to_dict()/chat_data, indent=2/4)` should do the trick.

Reply:
We can do it for the update, but not for user/chat data, since it can be not json serializable. We can use pprint, but I don't like how it looks, and the huge indents that can quickly consume the 4096 message length limit. In the example chat data is empty and I imagine it's usually a lot shorter than the update, so I suggest we leave it unformatted.

Reply:
Just noticed reason `.to_dict`/`.to_json` includes `_effective_*` for some reason.

Reply:
I see. I think I'll vote for `pprint` then. The length limit is covered by the comment IMO. And as the error handler should be informative, a flat representation of a dict doesn't fulfill that goal imo. If only one of the `chat_data` s values is a dict/list, it quickly becomes a headache to read that.
As for the private attributes in `to_dict()`: Thanks for opening an issue. Already commented. We can ignore that for this PR, though.